nodeup/pkg/model/bootstrap_client.go

```diff
@@ -45,14 +47,20 @@ func (b BootstrapClientBuilder) Build(c *fi.NodeupModelBuilderContext) error {
 	}
 
 	var authenticator bootstrap.Authenticator
+	var resolver resolver.Resolver
+
 	var err error
 	switch b.BootConfig.CloudProvider {
 	case kops.CloudProviderAWS:
 		authenticator, err = awsup.NewAWSAuthenticator(b.Cloud.Region())
 	case kops.CloudProviderGCE:
 		authenticator, err = gcetpmsigner.NewTPMAuthenticator()
-		// We don't use the custom resolver here in gossip mode (though we could);
-		// instead we use this as a check that protokube has now started.
+
+		discovery, err := gcediscovery.New()
+		if err != nil {
+			return err
+		}
+		resolver = discovery
```

Comment:
`gcetpmsigner.NewTPMAuthenticator` also returns error, so this would overwrite that. Move this below the error check for `authenticator`?

Reply:
Ouch - good call, thank you!